### PR TITLE
Remove CPU/memory limits

### PR DESCRIPTION
**Pull request by wtty-fool**
Towards https://github.com/giantswarm/giantswarm/issues/8569

### CODE REVIEW

**Comment by wtty-fool** [comment]
@kopiczko PTAL and see if this is the expected solution

File: helm/kube-state-metrics-app/templates/deployment.yaml [hunk 1]
```diff
@@ -40,9 +40,6 @@ spec:
       - name: addon-resizer
         image: "{{ .Values.image.registry }}/{{ .Values.imageResizer.name }}:{{ .Values.imageResizer.tag }}"
         resources:
-          limits:
```
Comment (sslavic) [inline helm/kube-state-metrics-app/templates/deployment.yaml]: @wtty-fool this PR removed resource limits from addon-resizer side-car, not from main container; was that intention?
addon-resizer based on configuration vertically scales main container https://github.com/kubernetes/autoscaler/tree/master/addon-resizer#addon-resizer

Comment (wtty-fool) [reply helm/kube-state-metrics-app/templates/deployment.yaml]: @sslavic No, I should have deleted the limits from the main container. I'll prepare a PR to sort this out.
Do you need me to prepare a PR for v.1.0.4 also?

Comment (wtty-fool) [reply helm/kube-state-metrics-app/templates/deployment.yaml]: Discussed on Slack. I know what to do now :)

**Pull request merged**